### PR TITLE
feat: implement gc command

### PR DESCRIPTION
**Pull request by thesayyn**
This is needed for a case where we use oci-layout as the registry storage directory and there are unreferenced blobs. We want to remove them during pull into the storage directory. I marked this as hidden to get some mileage in rules_oci.

### CODE REVIEW

File: cmd/crane/cmd/pull.go [hunk 1]
```diff
@@ -30,8 +30,8 @@ import (
 // NewCmdPull creates a new cobra.Command for the pull subcommand.
 func NewCmdPull(options *[]crane.Option) *cobra.Command {
 	var (
-		cachePath, format string
-		annotateRef       bool
+		cachePath, format  string
+		annotateRef, prune bool
```
Comment (imjasonh) [inline cmd/crane/cmd/pull.go:34]: I don't think it's appropriate to have a command for pulling be responsible for GCing. Maybe a separate hidden command that prunes an OCI layout directory would be appropriate, which you could run after pulling into that OCI layout directory.

It might make sense for layout.GC to move to an internal package, or at least have a comment saying it's experimental and shouldn't be relied upon outside of this path.

It'd also be great to see tests for it, since it's fairly complex and deletes data. There are some interesting corner cases in GCing that it would be great to have covered by tests.

Comment (thesayyn) [reply cmd/crane/cmd/pull.go:34]: hmm, i'd be okay with having this feature as a separate command. i believe in that case, we should name this `crane gc` and specifically say that it only supports local oci layouts

Comment (thesayyn) [reply cmd/crane/cmd/pull.go:34]: about corner cases, are there any specific cases you are aware of? i will add tests for basic cases but would be great to hear from you as well.

Comment (thesayyn) [reply cmd/crane/cmd/pull.go]: Done @imjasonh

Comment (thesayyn) [reply cmd/crane/cmd/pull.go]: Added some basic tests.

**Comment by thesayyn** [comment]
Test failures seems unrelated. probably expired token.

File: cmd/crane/cmd/pull.go (hunk 1 above)
Comment (thesayyn) [reply cmd/crane/cmd/pull.go]: ping @imjasonh

Comment (thesayyn) [reply cmd/crane/cmd/pull.go]: ping @jonjohnsonjr

**Review by imjasonh — APPROVED** [review]
